src/fixes/unordered_key.rs

```diff
@@ -0,0 +1,262 @@
+use super::Fix;
+use crate::common::*;
+
+pub(crate) struct UnorderedKeyFixer<'a> {
+    name: &'a str,
+}
+
+impl Default for UnorderedKeyFixer<'_> {
+    fn default() -> Self {
+        Self {
+            name: "UnorderedKey",
+        }
+    }
+}
+
+// When we sort the keys, we handle a significant line (with key) with all previous comments
+// as a whole. E. g.:
+// ```
+// B=C
+// # Comment
+// A=B
+// ```
+// will be fixed to:
+// ```
+// # Comment
+// A=B
+// B=C
+// ```
+// We check the order separately in each group of entries (groups are separated by blank lines).
+impl Fix for UnorderedKeyFixer<'_> {
+    fn name(&self) -> &str {
+        self.name
+    }
+
+    fn fix_warnings(
+        &self,
+        warnings: Vec<&mut Warning>,
+        lines: &mut Vec<LineEntry>,
+    ) -> Option<usize> {
+        // We find all sorting groups and sort them
+
+        let mut sorted_part_started = false;
+        let mut start = None;
+        let mut end = None;
+
+        for i in 0..lines.len() {
+            let line = &lines[i];
+            if line.is_empty() {
+                if sorted_part_started {
+                    sorted_part_started = false;
+                    Self::sort_part(
+                        &mut lines[start.expect("start index")..end.expect("end index")],
+                    );
+                }
+                start = None;
+            } else {
+                if start.is_none() {
+                    start = Some(i);
+                }
+                if !line.is_comment() {
+                    end = Some(i + 1);
+                    sorted_part_started = true;
+                }
+            }
+        }
+
+        // The last group
+        if sorted_part_started {
```

Comment:
What about just add in the body of the `for` cycle additional condition for last group instead of checking after the loop?
For example, in `for`:
```
if line.is_empty() || (i == lines.len() - 1)
```

Reply:
In this case, we need to add more conditions, since the `end` variable has not been assigned a value `lines.len() - 1`.

Reply:
Yeah, you're right, sorry :slightly_smiling_face:

Reply:
Please see [this comment](https://github.com/dotenv-linter/dotenv-linter/pull/261#discussion_r481022360), it solves this issue

Reply:
Please see [this comment](https://github.com/dotenv-linter/dotenv-linter/pull/261#discussion_r481022360), it solves this issue

Reply:
> In this case, we need to add more conditions, since the `end` variable has not been assigned a value `lines.len() - 1`.

Current tests cannot cover this condition anyway, they always have `\n` at the end. Please see also [comment](https://github.com/dotenv-linter/dotenv-linter/pull/261#discussion_r481040776)

Reply:
Done